src/modules/Modules.tsx

```diff
@@ -1,19 +1,23 @@
 import React from 'react';
+import { useLocation } from 'react-router-dom';
 
 import { AlertsManager } from 'core/api';
 import AuthProvider from 'core/auth';
 import PatternsProvider from 'core/patterns';
 import TechnologiesProvider from 'core/technologies';
 
+import { shouldLoadPatterns, shouldLoadTechnologies } from './shouldLoadDataOnInit';
 import ModulesRouter from './ModulesRouter';
 
 const Modules = () => {
+  const { pathname, search } = useLocation();
+
   return (
     <>
       <AlertsManager />
       <AuthProvider>
-        <PatternsProvider>
-          <TechnologiesProvider>
+        <PatternsProvider getOnMount={shouldLoadPatterns(pathname, search)}>
```

Comment:
We have one big issue with this approach - always we need to reset our state when we leave a dedicated route or loading flags/errors/data - because there is a bug connected to the admin panel. If you open the admin panel and leave from this route - spinner in templates searcher for patterns rotates. 

So how it should work:

- Patterns/Technologies provider exposes methods to get both entities list - also remove calls in componentDidMount,
- Get on mount code is completely removed - because this is over complicated and indirectly connected - which creates bugs which I mention above.
- In this files you have providers like before - use also all connected code for that,
- Go to modules router and call there methods from technologies/patterns providers in useEffect,
- Wrap with Providers your admin panel - with TechnologiesProvider and Patterns Provider - and use call there here this will create the additional component instance and overlap global Providers instance - so you will have independent states

Problem solved